### PR TITLE
I went on a CheckStyle rampage.

### PR DESCRIPTION
**Pull request by LadySerenaKitty**
### Contains
Lots and lots of checkstyle fixes!  I skipped over a couple types because I don't wanna break things.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine-tests/src/main/java/org/terasology/TerasologyTestingEnvironment.java [hunk 1]
```diff
@@ -104,12 +104,12 @@ public void setup() throws Exception {
         EntitySystemSetupUtil.addReflectionBasedLibraries(context);
         EntitySystemSetupUtil.addEntityManagementRelatedClasses(context);
         engineEntityManager = context.get(EngineEntityManager.class);
-        BlockManager blockManager = context.get(BlockManager.class);
+        BlockManager blockManagerLocal = context.get(BlockManager.class); // added Local to avoid HiddenFieldCheck
```
Comment (skaldarnar) [inline engine-tests/src/main/java/org/terasology/TerasologyTestingEnvironment.java]: Is the local BlockManager even needed? It actually be set up in `@BeforeClass` ...

**Comment by skaldarnar** [comment]
@GooeyHub ok to test

For the next time I advise to fix the warnings at least in different commits. 74 edited files are a lot for review, and it may sometimes not be clear what a change is all about. Try to work on it in a more structured way, e.g., fix all unused imports first, commit with message `"Fixed unused imports"`, and go on to the next item.

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/527/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java [hunk 2]
```diff
@@ -133,8 +133,8 @@ public boolean step() {
         context.get(ComponentSystemManager.class).register(new RelevanceSystem(chunkProvider), "engine:relevanceSystem");
         EntityAwareWorldProvider entityWorldProvider = new EntityAwareWorldProvider(
                 new WorldProviderCoreImpl(worldInfo, chunkProvider, blockManager.getBlock(BlockManager.UNLOADED_ID),
-                        context)
-                , context);
+                        context),
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java]: Probably just a question of personal taste (?):
```Java
EntityAwareWorldProvider entityWorldProvider = new EntityAwareWorldProvider(new WorldProviderCoreImpl(
    worldInfo, chunkProvider, blockManager.getBlock(BlockManager.UNLOADED_ID), context), context);
```

File: engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java [hunk 3]
```diff
@@ -125,17 +125,17 @@ private void loop(EntityRef entity, LocationComponent location,
 
             // if our AI is aggressive or hunter go and hunt player else run away
             // if wild
-            if (ai.aggressive) {
+            /*if (ai.aggressive) {
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java]: I think that just commenting here is not the best solution. You could either
- try to fix the code (if possible)
- re-structure and organize the comments and TODOs into a proper task description and pseudocode solution. this should probably be an issue on its own, referring to the legacy code

File: engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java (hunk 2 above)
Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java]: This was because the line was too long (>175).

File: engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java (hunk 3 above)
Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java]: At this point, CheckStyle was complaining about an empty block.  The code was:
`if (ai.aggressive) {
}`

File: engine-tests/src/main/java/org/terasology/TerasologyTestingEnvironment.java (hunk 1 above)
Comment (LadySerenaKitty) [inline engine-tests/src/main/java/org/terasology/TerasologyTestingEnvironment.java]: This was an instance of `HiddenFieldCheck`, with CheckStyle note `'blockManager' hides a field.`

File: engine-tests/src/test/java/org/terasology/persistence/WorldSerializerTest.java [hunk 4]
```diff
@@ -50,7 +35,8 @@
     public void testNotPersistedIfFlagedOtherwise() throws Exception {
         EngineEntityManager entityManager = context.get(EngineEntityManager.class);
         EntityBuilder entityBuilder = entityManager.newBuilder();
-        WorldSerializer worldSerializer = new WorldSerializerImpl(entityManager, new PrefabSerializer(entityManager.getComponentLibrary(), entityManager.getTypeSerializerLibrary()));
+        WorldSerializer worldSerializer =
+                new WorldSerializerImpl(entityManager, new PrefabSerializer(entityManager.getComponentLibrary(), entityManager.getTypeSerializerLibrary()));
```
Comment (msteiger) [inline engine-tests/src/test/java/org/terasology/persistence/WorldSerializerTest.java]: I suggest extracting a local variable for the prefab serializer. This makes is easier to read and the line become significantly shorter.

File: engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java [hunk 5]
```diff
@@ -51,11 +51,11 @@ public NoiseTest(Noise noiseGen) {
     @Parameters(name = "{0}")
     public static Collection<Noise[]> data() {
         return Arrays.asList(new Noise[][]{
-                { new WhiteNoise(0xCAFE) },
-                { new DiscreteWhiteNoise(0xCAFE) },
-                { new SimplexNoise(0xCAFE) },
-                { new PerlinNoise(0xCAFE) },
-                { new BrownianNoise(new WhiteNoise(0xCAFE), 3) }
```
Comment (msteiger) [inline engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java]: I think that the checkstyle rule should be changed here instead. The rule currently treats all opening brackets equals, but I'd restrict it to classes and method (or maybe trailing whitespace in general instead?).

File: engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java (hunk 2 above)
Comment (msteiger) [inline engine/src/main/java/org/terasology/engine/modes/loadProcesses/InitialiseWorld.java]: Again, I'd rather extract local variables here. The `WorldProviderCoreImpl`, the unloaded block instance, etc.

File: engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java (hunk 5 above)
Comment (msteiger) [inline engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java]: This should be the rule: https://github.com/MovingBlocks/TeraConfig/blob/master/checkstyle/checkstyle.xml#L138-L140

File: engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java [hunk 6]
```diff
@@ -172,6 +172,9 @@ public Matrix4f getInverseViewProjectionMatrix() {
         return null;
     }
 
+    /**
+     * @deprecated
```
Comment (msteiger) [inline engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java]: It would be quite helpful to know why this is deprecated and what to use instead. If you don't know, you might want to leave this as is, so somebody who does might stumble over it in the future and comment this properly.

**Comment by msteiger** [comment]
Looks almost good to me, just a few minor issues. Thanks - code quality 4tw!

File: engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java (hunk 6 above)
Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java]: I would like to know this, too.  Unfortunately, the codebase doesn't seem to explain these deprecations very well.  I think I'm going to file a BR for these deprecations, because not everyone looks at CheckStyle results, anyway.

Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java]: I agree with Martin, you might fix the checkstyle warning this way but you'd also hide the longer term issue of eliminating these methods altogether.

File: engine/src/main/java/org/terasology/rendering/opengl/PostProcessor.java [hunk 7]
```diff
@@ -121,13 +121,13 @@
...
     private Buffers buffers = new Buffers();
 
-    ThreadManager threadManager = CoreRegistry.get(ThreadManager.class);
-
     private boolean isTakingScreenshot;
 
     private RenderingConfig renderingConfig = CoreRegistry.get(Config.class).getRendering();
     private RenderingDebugConfig renderingDebugConfig = renderingConfig.getDebug();
 
+    ThreadManager threadManager = CoreRegistry.get(ThreadManager.class);
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/opengl/PostProcessor.java]: Hmmmm... I have the funny feeling that this was in the right place, but should be private. I don't see why it should be package-private. Can you make it private and see if anything explodes?

File: engine-tests/src/test/java/org/terasology/persistence/internal/StorageManagerTest.java [hunk 8]
```diff
@@ -129,9 +127,9 @@ public void setup() throws Exception {
         testBlock2 = context.get(BlockManager.class).getBlock("test:testblock2");
 
         context.put(ChunkProvider.class, mock(ChunkProvider.class));
-        BiomeManager biomeManager = mock(BiomeManager.class);
-        when(biomeManager.getBiomes()).thenReturn(Collections.<Biome>emptyList());
-        context.put(BiomeManager.class, biomeManager);
+        BiomeManager biomeManagerLocal = mock(BiomeManager.class);
```
Comment (emanuele3d) [inline engine-tests/src/test/java/org/terasology/persistence/internal/StorageManagerTest.java]: This is a mock object used for testing purposes. Call it for what it is! "mockBiomeManager"!

File: engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java (hunk 5 above)
Comment (emanuele3d) [inline engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java]: @msteiger: to be clear, are you suggesting LadySerenaKitty to revert this change and modify the checkstyle rule right in this PR?

File: engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java (hunk 3 above)
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java]: Hmmmm... ideally here we'd track down who commented out the content of the if block. I wonder if there's an easy way to do that via git.

File: engine-tests/src/test/java/org/terasology/persistence/internal/StorageManagerTest.java (hunk 8 above)
Comment (LadySerenaKitty) [inline engine-tests/src/test/java/org/terasology/persistence/internal/StorageManagerTest.java]: Haha, okay.

File: engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java [hunk 9]
```diff
@@ -21,18 +21,12 @@
...
-    FEATURE_USE_MATRIX_STACK
-            (0b00100000),
-    FEATURE_USE_FORWARD_LIGHTING
-            (0b01000000);
+    FEATURE_REFRACTIVE_PASS (0b00000001),
+    FEATURE_ALPHA_REJECT (0b00000010),
+    FEATURE_LIGHT_POINT (0b00000100),
+    FEATURE_LIGHT_DIRECTIONAL (0b00001000),
+    FEATURE_USE_MATRIX_STACK (0b00100000),
+    FEATURE_USE_FORWARD_LIGHTING (0b01000000);
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: I wonder if it would be horribly inappropriate here to line up (with spaces) the binary codes. @Cervator? I seem to remember we had a discussion about spaces/tabs once. What do you think about this case?

File: engine/src/main/java/org/terasology/rendering/backdrop/BackdropProvider.java [hunk 10]
```diff
@@ -24,8 +24,6 @@
  * This is generally intended to be the sky, but the theatrical term "backdrop" is used to include
  * anything in the background. On the other hand, the term "background" itself was avoided as it is
  * used in the context of background/foreground processes and threads.
- *
- * Created by manu on 13.01.2015.
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/backdrop/BackdropProvider.java]: My ego feels shattered, but that's a good thing. =D

File: engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java (hunk 6 above)
Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/rendering/cameras/OculusStereoCamera.java]: Okay, I'll go back and remove the javadoc annotations.

File: modules/Core/src/test/java/org/terasology/world/generator/TreeTests.java [hunk 11]
```diff
@@ -126,9 +126,9 @@ public Block setBlock(int x, int y, int z, Block block) {
             };
 
             Random random = new MersenneRandom(seed);
-            BlockManager blockManager = CoreRegistry.get(BlockManager.class);
+            BlockManager blockManagerLocal = CoreRegistry.get(BlockManager.class);
```
Comment (emanuele3d) [inline modules/Core/src/test/java/org/terasology/world/generator/TreeTests.java]: Only slightly different situation from the biomeManagerLocal comment earlier: in this case here "blockManager" is appropriate, because the one stored in the CoreRegistry is THE blockManager instance. However, the class-level field "blockManager" and "biomeManager" would be better named mockBlockManager and mockBiomeManager because they are mock objects.

**Comment by emanuele3d** [comment]
Just wanted to thank you @LadySerenaKitty for going through all these CheckStyle warnings. Please address the few issues still standing and this PR will be good to go.

File: engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java (hunk 3 above)
Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java]: I'd say use the blame command.  I wonder if you can do version-by-version blames on GitHub.

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/533/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java (hunk 3 above)
Comment (Cervator) [inline engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java]: Probably it was me slowly tearing out aging pieces originally introduced by Miniion and/or the ancient gelcubes but superseded but later changes. I don't think the functionality is used any more and can be removed rather than commented out.

File: engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java (hunk 5 above)
Comment (Cervator) [inline engine-tests/src/test/java/org/terasology/utilities/NoiseTest.java]: Probably just adjust what warnings are fixed and worry about rule changes separately (leaving the affected warnings alone for now)

File: engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java (hunk 9 above)
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: Historically we've avoided it. Probably from some sort of "hard to maintain" or keep consistent angle, maybe in part from being unable to write style rules that can consider lined up elements (may end up getting complaints about excessive spaces)

File: engine/src/main/java/org/terasology/rendering/backdrop/BackdropProvider.java (hunk 10 above)
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/backdrop/BackdropProvider.java]: :fireworks: :heart: :cry: :person_frowning: :+1:

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/535/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java [hunk 12]
```diff
@@ -21,11 +21,11 @@
...
-    FEATURE_REFRACTIVE_PASS (0b00000001),
-    FEATURE_ALPHA_REJECT (0b00000010),
-    FEATURE_LIGHT_POINT (0b00000100),
-    FEATURE_LIGHT_DIRECTIONAL (0b00001000),
-    FEATURE_USE_MATRIX_STACK (0b00100000),
+    FEATURE_REFRACTIVE_PASS      (0b00000001),
+    FEATURE_ALPHA_REJECT         (0b00000010),
+    FEATURE_LIGHT_POINT          (0b00000100),
+    FEATURE_LIGHT_DIRECTIONAL    (0b00001000),
+    FEATURE_USE_MATRIX_STACK     (0b00100000),
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: @Cervator: can you double check on this? Is it ok to add some spaces this way? We discussed these kind of things once and in general it wasn't a good idea to add formatting spaces. Is this a good exception to the rule?

Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: @emanuele3d You said you wanted thingies to be lined up.  Besides, it is more clear this way.

Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: @LadySerenaKitty: I know, and I personally agree it's clearer. But I'd like @Cervator's greenlight on this as we did discuss the issue of formatting spaces and he mentioned in general they aren't a good idea. Perhaps this is a good exception. Let's just wait and see what he says.

Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: I actually commented earlier, but it sounds like in the time it took me to review and comment this change was made and my comment got swallowed as having been made against an obsolete commit :-)

Historically we've avoided space-lining up things as it is hard to enforce consistency and maybe even impossible to code rules for it (one tool may warn about the extra whitespace while another ignores it). It may also end up disrupted again later.

Fundamentally I don't mind and we can leave it in, it just may trigger other whitespace style warnings somewhere.

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/537/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java (hunk 9 above)
Comment (msteiger) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: I think that this is a special case and aligning the bit fields makes sense here. CheckStyle should not complain with the current settings.

File: engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java (hunk 12 above)
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: Wouldn't in this case the old formatting be a readable alternative?

Comment (msteiger) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: I think that the new version is easier to read.

Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: Readability is everything!  All developers must be able to **easily** read _ALL THE THINGS_.

Comment (skaldarnar) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: @msteiger the one with space-aligned arguments? Checkstyle not complaining about it?

Comment (LadySerenaKitty) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: @skaldarnar Yea, Checkstyle is complaining about it. 😿

Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/assets/shader/ShaderProgramFeature.java]: I knew it. I knew this would be a can of worms. =P

I recommend to revert the change then, leaving no spaces and no alignment. Not as pretty but I'd rather not create a new precedent where CheckStyle is ignored.

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/543/
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
I finally found a chance to check it locally, with a few minutes to spare to apply a few final tiny tweaks on top (https://github.com/MovingBlocks/Terasology/commit/104580eb5cf222b0448fde6c790302b2571ccf76) - no explosions occurred! Other than the expected ones, anyway :-)

I'm sure we could find more nits to pick but that can be the next PR!

@LadySerenaKitty thanks a lot! :cat2:
